### DISCUSSION TITLE
connector.getProvider({ chainId }) not working

### DISCUSSION THREAD

**Discussion by kombos** [wevm/wagmi#1957] [General]
I have an app that needs to connect to both polygon and ethereum. currently while connected to ethereum, i need to create a polygon transaction, which requires a provider object from connector, configured to polygon's chain parameters (chain id). for that I'm doing this: 

```
const polygonProvider = await connector.getProvider({
    chainId: 137,
  })
```

this doesnt seem to work, since on logging the provider object, the chain ID value shown is "0x1" (ethereum mainnet). how to get the polygon chain's provider, without switching the network? 

npm version :      "wagmi": "^0.11.5"

**Comment by tmm** [comment]
What Connector are you using? This feature is not supported by all Connectors. If you are sending a transaction, you are better off using [`useSendTransaction`](https://wagmi.sh/react/hooks/useSendTransaction) (or [`sendTransaction`](https://wagmi.sh/core/actions/sendTransaction)) and validating the current chain id ahead of time.

**Reply by kombos** [reply]
> What Connector are you using? This feature is not supported by all Connectors. If you are sending a transaction, you are better off using [`useSendTransaction`](https://wagmi.sh/react/hooks/useSendTransaction) (or [`sendTransaction`](https://wagmi.sh/core/actions/sendTransaction)) and validating the current chain id ahead of time.

i'm connected to Metamask using [web3Modal library (of WalletConnect)](https://web3modal.com/), which internally uses wagmi hooks. also I require to get the 'provider' from connector because i need to provide it to [posClient of maticjs](https://wiki.polygon.technology/docs/develop/ethereum-polygon/matic-js/pos/), the offical sdk of Polygon PoS used to do polygon pos bridge related transactions. This posClient takes two providers (parent and child provider), one for ethereum (chain id 1)  and other for Polygon (chain id 137). how can i get these two instances of providers?

**Reply by tmm** [reply]
Sounds like you want the [`getProvider`](https://wagmi.sh/core/actions/getProvider) action.

**Reply by kombos** [reply]
> Sounds like you want the [`getProvider`](https://wagmi.sh/core/actions/getProvider) action.

I tried the getProvider, giving the chainId as argument, and it sets the chainId context correctly, but it returns a StaticJsonRpcProvider type. This is different from the provider type that is returned by connector.getProvider(). turns out that the posClient of maticjs is compatible with the provider type returned by connector.getProvider()... any way we can convert the StaticJsonRpcProvider type to the one similar to connector.getProvider() type (or web3provider type) ?

**Reply by kombos** [reply]
any ideas on this @tmm  ? how to convert the provider type (from useProvider or getProvider)  to something similar to connector.getProvider() type, like a Wallet type (of ethers), or web3provider type ?
